### PR TITLE
Restructured 3rd-party-licenses.txt and rephrased corresponding section in README.md

### PR DESCRIPTION
**Pull request by ralph-lange**

### CODE REVIEW

File: 3rd-party-licenses.txt [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
+
+--------------------------------------------------------------------------
+Overview
+--------------------------------------------------------------------------
+
+Multiple folders contain copies of the project Lab Project FreeRTOS POSIX,
+licensed under the MIT license.
+
+Name:      Lab Project FreeRTOS POSIX
+Version:   <<fill version number of commit id>>
```
Comment (ralph-lange) [inline 3rd-party-licenses.txt:17]: @pablogs9, can you please provide the version number?

File: 3rd-party-licenses.txt [hunk 2]
```diff
@@ -1,8 +1,40 @@
...
+Copyright: Copyright (c) 2019 FreeRTOS
+
+
+The folders microros_nucleo_f446ze_extensions and
+microros_olimex_e407_extensions contain code generated by the
+STM32CubeMX software tool of STMicroelectronics. The generated code is
+licensed by ST under the Ultimate Liberty license SLA0044.
+
+Name:      Code generated by STM32CubeMX
+Version:   <<fill version number of the STM32CubeMX tool>>
```
Comment (ralph-lange) [inline 3rd-party-licenses.txt:29]: @pablogs9, can you please provide the version number?

File: 3rd-party-licenses.txt (hunk 1 above)
Comment (pablogs9) [reply 3rd-party-licenses.txt]: Done

File: 3rd-party-licenses.txt (hunk 2 above)
Comment (pablogs9) [reply 3rd-party-licenses.txt]: Done

**Comment by pablogs9** [comment]
Hello @ralph-lange squash-merge if you are ok with my changes

**Pull request merged**